src/test/java/teammates/logic/core/InstructorsLogicTest.java

```diff
@@ -646,4 +647,26 @@ private void testGetCoOwnersForCourse() {
                 && generatedCoOwnersEmails.containsAll(coOwnersEmailsFromDataBundle));
     }
 
+    private void testUpdateToEnsureValidityOfInstructorsForTheCourse() {
+        ______TS("Should not grant the currently being edited instructor the privilege of modifying instructors");
+        String courseId = "idOfTypicalCourse1";
+        InstructorAttributes instructorToUpdate =
+                InstructorAttributes.builder(courseId, "idOfInstructor4@gmail.com")
+                        .withGoogleId("idOfInstructor4")
+                        .withPrivileges(
+                                new InstructorPrivileges(
+                                        Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_TUTOR
+                                )
+                        ).build();
+        instructorsLogic.updateToEnsureValidityOfInstructorsForTheCourse(courseId, instructorToUpdate);
+
+        assertFalse(instructorToUpdate.privileges.isAllowedForPrivilege(Const.InstructorPermissions.CAN_MODIFY_INSTRUCTOR));
+
+        ______TS("Should grant the currently being edited instructor the privilege of modifying instructors");
+        courseId = "idOfCourseNoEvals";
+        instructorsLogic.updateToEnsureValidityOfInstructorsForTheCourse(courseId, instructorToUpdate);
+
+        assertTrue(instructorToUpdate.privileges.isAllowedForPrivilege(Const.InstructorPermissions.CAN_MODIFY_INSTRUCTOR));
```

Comment:
This is insufficient. Looking at the logic that determines whether the instructor needs to be updated or not:
```
        boolean isLastRegInstructorWithPrivilege = numOfInstrCanModifyInstructor <= 1
                && instrWithModifyInstructorPrivilege != null
                && (!instrWithModifyInstructorPrivilege.isRegistered()
                || instrWithModifyInstructorPrivilege.googleId
                .equals(instructorToEdit.googleId));
```
we should expect a few different test cases, particularly on the negative side.

Even the current one is not well-explained. Why is this instructor granted the permission here, and not on the previous test?

Reply:
We would need more test data for this right? Most courses in the `typicalDataBundle.json` has more than 1 instructor with modifying instructor privilege

Reply:
You can create new courses, new instructors, or deleting existing instructors without touching the test data